src/Engine/EmitMSIL/Replication.cs

```diff
@@ -199,12 +200,25 @@ public static object ReplicationLogic(List<CLRFunctionEndPoint> feps, IList args
             //Turn the replication guides into a guide -> List args data structure
             var partialInstructions = Replicator.BuildPartialReplicationInstructions(partialReplicationGuides);
 
+            // TODO_MSIL: Figure it the best way to interpret/pass around the this pointer.
+            CLRStackValue thisPtr = CLRStackValue.Null;
+            bool staticFuncCall = feps.All(x => x.IsStatic);
```

Comment:
is this existing logic or new - couldn't we have two candidate feps in the same class, one static and one not?

Reply:
THis is new logic.
I do not think we can have mixed static/non-static function endpoints at the same time. 

Seems to me like:
Function calls ClassName.FunctionName should only resolve to static functions.
Function calls classInstance.Function should only resolve to non-static functions.
Please correct me if I am wrong. @mjkkirschner @aparajit-pratap 

I guess static-ness vs non-static-ness should be something we figure out at the compile stage. But it seemed that we could infer it at runtime replication stage too.

Reply:
I do not mean AST.
I mean the design script language. 
I do not think we will be allowed to call static methods from a class instance. Similarly we will not be allowed to call instance methods from the type name.